var/spack/repos/builtin/packages/upp/package.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Upp(CMakePackage):
+    """
+    The Unified Post Processor (UPP) software package is a software
+    package designed to generate useful products from raw model
+    output.
+    """
+
+    homepage = "https://github.com/NOAA-EMC/UPP"
+    git = "https://github.com/NOAA-EMC/UPP.git"
+
+    maintainers = ['kgerheiser', 'edwardhartnett', 'Hang-Lei-NOAA']
+
+    version('10.0.9', branch='upp_v10.0.9', submodules=True)
```

Comment:
The general practice for numbered branches is to use the `tag` or, less often, the commit.  The former would look like:

```suggestion
    version('10.0.9', tag='upp_v10.0.9', submodules=True)
```